### DISCUSSION TITLE
How to use Django daphne and socketio

### DISCUSSION THREAD

**Discussion by sh1man999** [miguelgrinberg/python-socketio#1349] [Q&A]
I have seen an example of using django + wsgi. But I haven't seen an example of using django + asgi. My application is asynchronous and runs with django + daphne + channels. i would like to use socketio. How do i do this, your recommendations and does it make sense to do this ?

**Comment by miguelgrinberg** [comment]
I haven't created an example specifically for Django, but the `ASGIApp` wrapper in this project should work well to integration Socket.IO in ASGI mode with any other ASGI app that you use.

**Comment by sh1man999** [comment]
could you write a small example ? Which web server should I use daphne or uvicorn ?

**Reply by miguelgrinberg** [reply]
You can use your favorite ASGI web server. I have used uvicorn with good results, but others whould work as well.

**Comment by sh1man999** [comment]
Is this relevant to the use case ?

https://psami.hashnode.dev/using-socketio-with-django-rest-framework-and-django-asgi-application

**Reply by miguelgrinberg** [reply]
This tutorial looks good to me. Very similar to the example I have for WSGI, but adapted to use ASGI.
